File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/parse_product_catalog_json-chatgpt.py
import json

def parse_product_catalog():
    json_data = """
    {
        "products": [
            {
                "id": 1,
                "name": "Laptop",
                "price": 999.99
            },
            {
                "id": 2,
                "name": "Mouse",
                "price": 29.99
            },
            {
                "id": 3,
                "name": "Keyboard",
                "price": 49.99
            }
        ]
    }
    """

    catalog = json.loads(json_data)

    for product in catalog["products"]:
        print(f"Product ID: {product['id']}")
        print(f"Name: {product['name']}")
        print(f"Price: ${product['price']}")
        print("-" * 30)

if __name__ == "__main__":
    parse_product_catalog()

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/ParseProductCatalogJSON-chatgpt.java
import org.json.JSONArray;
import org.json.JSONObject;

public class ProductCatalogParser {

    public static void main(String[] args) {

        String jsonData = """
        {
          "products": [
            {
              "id": 1,
              "name": "Laptop",
              "price": 999.99
            },
            {
              "id": 2,
              "name": "Mouse",
              "price": 29.99
            },
            {
              "id": 3,
              "name": "Keyboard",
              "price": 49.99
            }
          ]
        }
        """;

        JSONObject catalog = new JSONObject(jsonData);
        JSONArray products = catalog.getJSONArray("products");

        for (int i = 0; i < products.length(); i++) {
            JSONObject product = products.getJSONObject(i);

            int id = product.getInt("id");
            String name = product.getString("name");
            double price = product.getDouble("price");

            System.out.println("Product ID: " + id);
            System.out.println("Name: " + name);
            System.out.println("Price: $" + price);
            System.out.println("-------------------------");
        }
    }
}